pkg/kudoctl/cmd/upgrade.go

```diff
@@ -34,16 +34,14 @@ var (
 )
 
 type options struct {
+	install.RepositoryOptions
```

Comment:
Nice usage of embedded types. Out of curiosity: what is your reasoning to embed a struct vs. making it a field e.g. `Repo install.RepositoryOptions`?

Reply:
this example is too simple to see it yet... but the flatter structure is easy to work with (although involves some magic:( ).   I was thinking of having grouping of common options (for instance options needed for working with repos, options needed for working with k8s).   Then embed those options into the options for your command.  I was further thinking of having a function for each of these structs for flag setup... thereby guaranteeing that cli messages are consistent.   This would result in outer structure function calling the inner... I'm not sure if it adds too much complexity for what it gains.   thoughts welcome 🙏